src/main/features/linux/mpris.ts

```diff
@@ -95,76 +103,69 @@ ipcMain.on('mpris-update-seek', (_event, arg) => {
-                : null;
-
-            mprisPlayer.metadata = {
-                'mpris:artUrl': upsizedImageUrl,
-                'mpris:length': song.duration ? Math.round((song.duration || 0) * 1e6) : null,
-                'mpris:trackid': song?.id
-                    ? mprisPlayer.objectPath(`track/${song.id?.replace('-', '')}`)
-                    : '',
-                'xesam:album': song.album || null,
-                'xesam:albumArtist': song.albumArtists?.length ? song.albumArtists[0].name : null,
-                'xesam:artist':
-                    song.artists?.length !== 0
-                        ? song.artists?.map((artist: RelatedArtist) => artist.name)
-                        : null,
-                'xesam:discNumber': song.discNumber ? song.discNumber : null,
-                'xesam:genre': song.genres?.length
-                    ? song.genres.map((genre: any) => genre.name)
-                    : null,
-                'xesam:title': song.name || null,
-                'xesam:trackNumber': song.trackNumber ? song.trackNumber : null,
-                'xesam:useCount':
-                    song.playCount !== null && song.playCount !== undefined ? song.playCount : null,
-            };
-        } catch (err) {
-            console.log(err);
+ipcMain.on('update-volume', (_event, volume) => {
+    mprisPlayer.volume = Number(volume) / 100;
+});
+
+const REPEAT_TO_MPRIS: Record<PlayerRepeat, string> = {
+    [PlayerRepeat.ALL]: 'Playlist',
+    [PlayerRepeat.ONE]: 'Track',
+    [PlayerRepeat.NONE]: 'None',
+};
+
+ipcMain.on('update-repeat', (_event, arg: PlayerRepeat) => {
+    mprisPlayer.loopStatus = REPEAT_TO_MPRIS[arg];
+});
+
+ipcMain.on('update-shuffle', (_event, shuffle: boolean) => {
+    mprisPlayer.shuffle = shuffle;
+});
+
+ipcMain.on('update-song', (_event, args: SongUpdate) => {
+    const { song, status, repeat, shuffle } = args || {};
+
+    try {
+        mprisPlayer.playbackStatus = status === PlayerStatus.PLAYING ? 'Playing' : 'Paused';
+
+        if (repeat) {
+            mprisPlayer.loopStatus = REPEAT_TO_MPRIS[repeat];
+        }
+
+        if (shuffle) {
+            mprisPlayer.shuffle = shuffle;
         }
-    },
-);
+
+        if (!song) return;
+
+        const upsizedImageUrl = song.imageUrl
+            ? song.imageUrl
+                  ?.replace(/&size=\d+/, '&size=300')
+                  .replace(/\?width=\d+/, '?width=300')
+                  .replace(/&height=\d+/, '&height=300')
+            : null;
+
+        mprisPlayer.metadata = {
+            'mpris:artUrl': upsizedImageUrl,
+            'mpris:length': song.duration ? Math.round((song.duration || 0) * 1e6) : null,
+            'mpris:trackid': song?.id
+                ? mprisPlayer.objectPath(`track/${song.id?.replace('-', '')}`)
+                : '',
+            'xesam:album': song.album || null,
+            'xesam:albumArtist': song.albumArtists?.length ? song.albumArtists[0].name : null,
```

Comment:
Why is `albumArtist` being condensed to just one artist? [The spec](https://www.freedesktop.org/wiki/Specifications/mpris-spec/metadata/#xesam:albumartist) specifies that it can be an array of strings.

```suggestion
            'xesam:albumArtist': song.albumArtists?.length
              ? song.albumArtists.map((artist: RelatedArtist => artist.name) : null,
```